### DISCUSSION TITLE
Raphtory Type Stubs!

### DISCUSSION THREAD

**Discussion by hallofstairs** [Pometry/Raphtory#1681] [Show and tell]
Hey everyone,

I came across `raphtory` a bit ago, and was disappointed that there weren't any type hints. With type hints, your IDE can help you catch errors early and provide better suggestions while you code.

So, I just made a PyPI library with Raphtory type stubs! 

Disclaimer: it's still a WIP, so it only covers (most of) the Graph Core API right now, and some of the types might be incomplete and/or incorrect. I also confirmed it works on VSCode, but not sure about other IDEs.

Install it via pip:

```bash
pip install raphtory-stubs
```

Try it out and let me know what you think! If you run into issues or have ideas for improvements, lmk on the [GitHub repo](https://github.com/hallofstairs/raphtory-stubs) and open an issue or PR.

Thanks!

**Comment by miratepuffin** [comment]
Hi @hallofstairs, thank you for making this! Would you have some time for a quick chat about how we can integrate/support this as a core part of Raphtory? As it is indeed annoying that this information doesn't come up in an IDE.

**Reply by hallofstairs** [reply]
Yeah totally! Can discuss on Twitter DM @hallofstairs
